client/windows/GUIClasses.cpp

```diff
@@ -507,14 +511,42 @@ CTavernWindow::CTavernWindow(const CGObjectInstance * TavernObj, const std::func
 		CCS->videoh->open(townObj->town->clientInfo.tavernVideo);
 	else
 		CCS->videoh->open(VideoPath::builtin("TAVERN.BIK"));
+
+	addInvite();
+}
+
+void CTavernWindow::addInvite()
+{
+	OBJECT_CONSTRUCTION_CAPTURING(255-DISPOSE);
+
+	if(VLC->settings()->getBoolean(EGameSettings::HEROES_INVITE))
```

Comment:
Perhaps use early return to avoid function that consist from single if block?
```suggestion
	if(!VLC->settings()->getBoolean(EGameSettings::HEROES_INVITE))
		return;
```